src/carousel/carousel.spec.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  iit,
+  it,
+  ddescribe,
+  describe,
+  expect,
+  inject,
+  injectAsync,
+  TestComponentBuilder,
+  beforeEachProviders
+} from 'angular2/testing';
+
+import {Component, NgFor} from 'angular2/angular2';
+
+import {NgbCarousel, NgbSlide} from './carousel';
+
+function getSlides(compiled) {
+  return compiled.querySelectorAll('.carousel-item');
+}
+
+function hasClass(element: Element, str: string): Boolean {
+  return new RegExp(`(^|\\s)${str}(\\s|$)`).test(element.className)
+}
+
+describe('ngb-carousel', () => {
+  describe('UI logic', () => {
+    let html: string;
+
+    beforeEach(() => {
+      html = `
+      <ngb-carousel>
+        <ngb-slide [(active)]="slides[0].active">
```

Comment:
IMO we should re-think the API we've got - `active` on each slide is not great since in theory it allows multiple active slides. 

How about an "active index" or similar on the `ngb-carousel` level?